omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/SessionController1_9.java

```diff
@@ -118,5 +125,34 @@ public void post(HttpServletRequest request, @RequestBody Map<String, String> bo
 	public void delete() {
 		Context.logout();
 	}
-	
+
+	/**
+	 * Get current provider
+	 *
+	 * @return Provider if the user is authenticated
+	 */
+	private Provider getCurrentProvider() {
+		UserContext userContext = Context.getUserContext();
+		Provider currentProvider = null;
+		if (userContext != null && userContext.getAuthenticatedUser() != null) {
+			User currentUser = userContext.getAuthenticatedUser();
+			Collection<Provider> providers = new HashSet<Provider>();
+			try {
+				Context.addProxyPrivilege(PrivilegeConstants.VIEW_PROVIDERS);
+				if (currentUser.getPerson() != null) {
+					providers = Context.getProviderService().getProvidersByPerson(currentUser.getPerson(), false);
+				}
+			}
+			finally {
+				Context.removeProxyPrivilege(PrivilegeConstants.VIEW_PROVIDERS);
+			}
+			if (providers.size() > 1) {
+				throw new IllegalStateException("Can't handle users with multiple provider accounts");
```

Comment:
It might be smarter here to to simply log an error but not throw an exception.  I think, at least to start, the majority of the users of this method will not care about the provider, and we wouldn't want to break everything just because a user had muliple provider accounts.

Reply:
You are right